nautobot_chatops/api/views/lookup.py

```diff
@@ -32,3 +33,26 @@ def get(self, request, *args, **kwargs):
             return HttpResponseNotFound(f"No {request.GET['grant_type']} {request.GET['name']} found")
 
         return JsonResponse(data={"value": value})
+
+
+class UserEmailLookupView(View):
+    """Look up a user_id by email."""
+
+    http_method_names = ["get"]
+
+    def get(self, request, *args, **kwargs):
+        """Handle an inbound GET request for a specific access grant value."""
+        for required_param in ("email", "platform"):
+            if required_param not in request.GET:
+                return HttpResponseBadRequest(f"Missing mandatory parameter {required_param}")
+
+        value = None
+        for dispatcher_class in Dispatcher.subclasses():
+            if dispatcher_class.platform_slug == request.GET["platform"]:
+                with contextlib.suppress(NotImplementedError):
+                    value = dispatcher_class.lookup_user_id_by_email(request.GET["email"])
+        return (
+            JsonResponse(data={"user_id": value})
+            if value
+            else HttpResponseNotFound(f"No user_id found for {request.GET['email']}")
+        )
```

Comment:
Proposing to simplify this by starting with

```
email = request.GET["email"]
if not email:
    ...
platform = request.GET["platform"]
...
```

Code will be better readable IMHO